### PR TITLE
Restore HuggingFace option

### PR DESCRIPTION
**Pull request by gheinrich**

### CODE REVIEW

File: hf_hub.py [hunk 1]
```diff
@@ -18,100 +18,111 @@
...
-
-hugginface_repo_mapping = {"RADIO": "nvidia/RADIO", "E-RADIO": "nvidia/E-RADIO"}
+from radio.hf_model import RADIOConfig, RADIOModel
 
 
 def main():
     """Main Routine.
 
-    Push a RADIO model to Hugging.
+    Construct and optionall push a RADIO model to Hugging.
```
Comment (mranzinger) [inline hf_hub.py:27]: Nit: Should be "optionally"

File: hf_hub.py [hunk 2]
```diff
@@ -18,100 +18,111 @@
...
     args = parser.parse_args()
 
     # Load the checkpoint and create the model.
     checkpoint = torch.load(args.checkpoint_path, map_location="cpu")
     model_args = checkpoint["args"]
 
     # Extract the state dict from the checkpoint.
-    state_dict = checkpoint["state_dict"]
+    if "state_dict_ema" in checkpoint:
+        state_dict = checkpoint["state_dict_ema"]
```
Comment (mranzinger) [inline hf_hub.py:64]: The EMA model is weird in that it never uses spectral reparametrization. In order to load this state dict, we should also do this:
```
model_args.spectral_reparam = False
```

File: hf_hub.py [hunk 3]
```diff
@@ -18,100 +18,111 @@
...
+        warning(f"Missing keys in state dict: {key_warn.missing_keys}")
+    if key_warn.unexpected_keys:
+        warning(f"Unexpected keys in state dict: {key_warn.unexpected_keys}")
 
     # Restore input conditioner.
     radio_model.input_conditioner.load_state_dict(
         get_prefix_state_dict(state_dict, "input_conditioner.")
     )
 
     radio_model.eval().cuda()
```
Comment (mranzinger) [inline hf_hub.py:90]: Are you planning on adding adaptor support?

File: radio/hf_model.py [hunk 4]
```diff
@@ -12,35 +12,56 @@
...
 
     def __init__(
         self,
         args: Optional[dict] = None,
-        version: Optional[str] = "v1",
-        return_summary: Optional[bool] = True,
-        return_spatial_features: Optional[bool] = True,
+        version: Optional[str] = DEFAULT_VERSION,
+        patch_size: Optional[int] = None,
+        max_resolution: Optional[int] = None,
```
Comment (mranzinger) [inline radio/hf_model.py:43]: Forgive my ignorance: Are these considered part of the config, or innately part of the model? It shouldn't be user configurable, but is this config allowing for that to be specified by the user?

File: hf_hub.py (hunk 1 above)
Comment (gheinrich) [reply hf_hub.py]: Fixed in latest commit thanks.

File: hf_hub.py (hunk 2 above)
Comment (gheinrich) [reply hf_hub.py:64]: Fixed in latest commit thanks.

File: radio/hf_model.py (hunk 4 above)
Comment (gheinrich) [reply radio/hf_model.py:43]: This is part of the config as we need it to instantiate the `RADIOModel`. We don't expect users to change it as the standard programming model fetches the config seamlessly:

```python
model = AutoModel.from_pretrained(hf_repo, trust_remote_code=True)
```

**Comment by gheinrich** [comment]
Thanks for the review, I tested the Torchhub updates on the `NVlabs/RADIO:dev/hf` repo using these example scripts:

* examples/knn_classification.py         
* examples/position_predictor.py               
* examples/visualize_features.py
* examples/zero_shot_imagenet.py

**Review by mranzinger — APPROVED** [review]
LGTM

**Pull request merged**